src/providers/v2/quote-provider.ts

```diff
@@ -87,7 +87,11 @@ export class V2QuoteProvider implements IV2QuoteProvider {
       for (const amount of amounts) {
         try {
           if (tradeType == TradeType.EXACT_INPUT) {
-            let outputAmount = amount.wrapped;
+            const firstPair = route.pairs[0]!;
+            let outputAmount = CurrencyAmount.fromRawAmount(
```

Comment:
I've been thinking if we can inline this special logic inside v2-sdk? 

```
private derivePercentAfterSellFees(inputAmount: CurrencyAmount<Token>): Percent {
    const sellFeeBips = firstPair.token0.equals(amount.wrapped.currency) ? firstPair.token0.wrapped.sellFeeBps : firstPair.token1.wrapped.sellFeeBps;
    if (sellFeeBips?.gt(BigNumber.from(0))) {
      return ONE_HUNDRED_PERCENT.subtract(new Percent(JSBI.BigInt(sellFeeBips)).divide(BASIS_POINTS))
    } else {
      return ZERO_PERCENT
    }
  }
```

seems cleaner this way, and we don't have to instantiate CurrencyAmount object one time just for this special case.